### ISSUE TITLE
Not building on non systemd systems

### ISSUE THREAD

**Issue by rapha8l** [liljencrantz/crush#28]
Hi again
I use void which uses runit and has not systemd libs, which leads to a building failure

> error: failed to run custom build command for `libsystemd-sys v0.2.2`
> 
> Caused by:
>   process didn't exit successfully: `/home/xxx/src/Shell/crush/target/debug/build/libsystemd-sys-04e3e7e70bffab3a/build-script-build` (exit code: 101)
> --- stdout
> `"pkg-config" "--libs" "--cflags" "libsystemd"` did not exit successfully: exit code: 1
> --- stderr
> Package libsystemd was not found in the pkg-config search path.
> Perhaps you should add the directory containing `libsystemd.pc'
> to the PKG_CONFIG_PATH environment variable
> No package 'libsystemd' found
> 
> 
> --- stderr
> thread 'main' panicked at 'systemd was not found via pkg-config nor via the env var LIBSYSTEMD_LDFLAGS', /home/xxx/.cargo/registry/src/github.com-1ecc6299db9ec823/libsystemd-sys-0.2.2/build.rs:19:13
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> 
> warning: build failed, waiting for other jobs to finish...
> error: build failed
>

**Comment by liljencrantz** [comment]
Agreed, it would be nice to make dbus and systemd into optional dependencies. Not sure how to do that, but I'll look into it.

Patches are welcome.

**Comment by rapha8l** [comment]
removing all references to systemd makes it build, though as systemd seems to be the way to store the journal, it gives

```
ls /
Error: Expected argument field to be of type Vec
Error: Failed to save history: No such file or directory (os error 2)
```
